pkg/core/rest/errors/error_handler.go

```diff
@@ -162,7 +162,9 @@ func HandleError(ctx context.Context, response *restful.Response, err error, tit
 		}
 	}
 	if ctx != nil {
-		if span := trace.SpanFromContext(ctx); span.IsRecording() {
+		span := trace.SpanFromContext(ctx)
+		span.RecordError(err, trace.WithStackTrace(true))
+		if span.IsRecording() {
```

Comment:
are you sure RecordError does not have to be inside `IsRecording` if?

Reply:
```
	// be set to Error, as this method does not change the Span status. If this
	// span is not being recorded or err is nil then this method does nothing.

```